apps/src/dance/Dance.js

```diff
@@ -176,16 +179,26 @@ Dance.prototype.init = function (config) {
 
   ReactDOM.render(
     <Provider store={getStore()}>
-      <AppView
-        visualizationColumn={
-          <DanceVisualizationColumn
-            showFinishButton={showFinishButton}
-            setSong={this.setSongCallback.bind(this)}
-            resetProgram={this.reset.bind(this)}
-          />
+      <ErrorBoundary
+        fallback={<ErrorFallbackPage />}
+        onError={(error, componentStack) =>
+          Lab2MetricsReporter.logError('Dance Party Error', error, {
```

Comment:
We should also setup this reporter by calling `Lab2MetricsReporter.updateCommonProperties` when the level first loads so we can set common fields like channel ID, app name, level ID, etc on payloads. Would be also good to make a similar note there about how we're using this reporter for now, but we should revisit and replace/refactor later.

Reply:
I wasn't 100% sure of the logic in this file, so I just opted to just get the current state and call `updatedProperties` inside the `onError` function to ensure everything's up to date.

Can/should it be moved up into `Dance.prototype.init`? AFAIK there's no harm in having it in `onError`, but will defer to preferences on location. Please double check that I loaded the right bits of data from the right places.